internal/backend/vars/vars.go

```diff
@@ -13,36 +14,53 @@ import (
 	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
 )
 
-type vars struct {
+type Vars struct {
 	db      db.DB
 	secrets secrets.Secrets
+	conns   sdkservices.Connections
 	z       *zap.Logger
 }
 
-func New(z *zap.Logger, db db.DB, secrets secrets.Secrets) sdkservices.Vars {
-	return &vars{db: db, z: z, secrets: secrets}
+func New(z *zap.Logger, db db.DB, secrets secrets.Secrets) *Vars {
+	return &Vars{db: db, z: z, secrets: secrets}
 }
 
 func varSecretKey(secret sdktypes.Var) string {
 	return fmt.Sprintf("%s/%s", secret.ScopeID().AsID().UUIDValue().String(), secret.Name())
 }
 
-func (v *vars) Set(ctx context.Context, vs ...sdktypes.Var) error {
+func (v *Vars) SetConnections(conns sdkservices.Connections) { v.conns = conns }
+
+func (v *Vars) Set(ctx context.Context, vs ...sdktypes.Var) error {
 	for i, va := range vs {
 		if va.IsSecret() {
 			key := varSecretKey(va)
 			if err := v.secrets.Set(ctx, key, va.Value()); err != nil {
-				//TODO: ENG-817 - handle dangling secrets in secret store
+				// TODO: ENG-817 - handle dangling secrets in secret store
 				return err
 			}
 
 			vs[i] = sdktypes.NewVar(va.Name(), key, true).WithScopeID(va.ScopeID())
 		}
 	}
-	return v.db.SetVars(ctx, vs)
+
+	if err := v.db.SetVars(ctx, vs); err != nil {
+		return err
+	}
+
+	sids := kittehs.ListToMap(vs, func(v sdktypes.Var) (sdktypes.VarScopeID, bool) { return v.ScopeID(), v.ScopeID().IsConnectionID() })
+	var errs []error
+	for sid, isConn := range sids {
+		if isConn {
+			if _, err := v.conns.RefreshStatus(ctx, sid.ToConnectionID()); err != nil {
+				errs = append(errs, err)
+			}
+		}
+	}
```

Comment:
this coupling of vars and connections  keeps getting bigger and bigger

Reply:
noted